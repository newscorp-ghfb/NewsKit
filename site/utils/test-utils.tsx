/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import {
  renderToFragmentWithThemeFactory,
  renderWithThemeFactory,
} from 'newskit/test/test-utils';
import {docsThemeLight} from '../theme/doc-theme';

export const renderWithTheme = renderWithThemeFactory(docsThemeLight);

export const renderToFragmentWithTheme: any = renderToFragmentWithThemeFactory(
  docsThemeLight,
);

export const mockNewsKitComponents = (...components: string[]) => () => ({
  ...(jest.requireActual('newskit') as any),
  ...components.reduce((acc, comp) => {
    acc[comp] = ({children, ...props}: any) => (
      <div
        data-comp={`NewsKit ${comp}`}
        data-testid={comp}
        data-props={JSON.stringify(props, null, 2)}
      >
        {children}
      </div>
    );
    return acc;
  }, {} as Record<string, Function>),
});
