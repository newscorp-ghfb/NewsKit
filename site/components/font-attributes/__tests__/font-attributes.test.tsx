import {renderToFragmentWithTheme} from 'newskit/test/test-utils';
import {FontAttributesProps} from '..';
import {FontAttributes} from '../font-attributes';

describe('FontAttributes', () => {
  test('renders default font attributes', () => {
    const fragment = renderToFragmentWithTheme(FontAttributes);
    expect(fragment).toMatchSnapshot();
  });

  test('renders with fontSize token', () => {
    const props: FontAttributesProps = {
      fontSizeToken: 'fontSize010',
    };
    const fragment = renderToFragmentWithTheme(FontAttributes, props);
    expect(fragment).toMatchSnapshot();
  });

  test('renders with fontWeightToken token', () => {
    const props: FontAttributesProps = {
      fontWeightToken: 'fontWeight010',
    };
    const fragment = renderToFragmentWithTheme(FontAttributes, props);
    expect(fragment).toMatchSnapshot();
  });

  test('renders with fontLineHeight token', () => {
    const props: FontAttributesProps = {
      fontLineHeight: 'fontLineHeight010',
    };
    const fragment = renderToFragmentWithTheme(FontAttributes, props);
    expect(fragment).toMatchSnapshot();
  });

  test('renders with fontLetterSpacing token', () => {
    const props: FontAttributesProps = {
      fontLetterSpacing: 'fontLetterSpacing030',
    };
    const fragment = renderToFragmentWithTheme(FontAttributes, props);
    expect(fragment).toMatchSnapshot();
  });
});
