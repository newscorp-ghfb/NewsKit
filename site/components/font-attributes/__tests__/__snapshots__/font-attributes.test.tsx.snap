// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FontAttributes renders default font attributes 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`FontAttributes renders with fontLetterSpacing token 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  letter-spacing: 0;
}

<div
    class="emotion-0"
  >
    The quick brown fox
  </div>
</DocumentFragment>
`;

exports[`FontAttributes renders with fontLineHeight token 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: 1;
}

<div
    class="emotion-0"
  >
    The quick brown fox jumps over the lazy dog
    <br />
     The quick brown fox jumps over the lazy dog 
  </div>
</DocumentFragment>
`;

exports[`FontAttributes renders with fontSize token 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 12px;
}

<div
    class="emotion-0"
  >
    The quick brown fox
  </div>
</DocumentFragment>
`;

exports[`FontAttributes renders with fontWeightToken token 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: 400;
}

<div
    class="emotion-0"
  >
    The quick brown fox
  </div>
</DocumentFragment>
`;
